### ISSUE TITLE
pathos for Python 3?

### ISSUE THREAD

**Issue by MichaelCurrie** [uqfoundation/pathos#77]
I would like to use `pathos.multiprocessing` so that I can get around the canonical [`multiprocessing`](https://docs.python.org/3/library/multiprocessing.html) package's use of `pickle`, [which can't handle pickling lambdas](http://stackoverflow.com/questions/19984152/what-can-multiprocessing-and-dill-do-together).  Apparently `dill` can, and `dill` is used in `pathos.multiprocessing`.

But I am using Python 3.  Here's what happens if I try `pip install pathos`:

```
ubuntu@ip-172-31-9-94:~/$ pip install pathos
Collecting pathos
  Downloading pathos-0.1a1.tgz (574kB)
    100% |████████████████████████████████| 577kB 989kB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-e8k_qu04/pathos/setup.py", line 66
        exec setup_code
                      ^
    SyntaxError: Missing parentheses in call to 'exec'
```

What's the word on getting this library to be Python 3 compatible?  Thanks!

**Comment by mmckerns** [comment]
@MichaelCurrie: `pathos` has been around since before python 3.x.  Most of it has been converted to python 3, however the portion that enables ssh-tunneling and ssh pipes has not been converted yet.  So, `pathos` is not yet a python 3 package.

However, there is good news.  If you are looking to use `pathos.multiprocessing` in python 3, you can.  You will only need the `multiprocess` package, which is what is underneath `pathos.multiprocessing`.  With it, you can pickle lambdas and other objects in parallel by leveraging `dill`.

**Comment by hallabalooza** [comment]
**@mmckerns:** Hi. The problem isn't pathos. The problem is that the setup scripts aren't compatible with Python 3. So there is no easy way to install pathos for Python 3.

**Comment by mmckerns** [comment]
Slight update: The remaining conversion is finally underway.  As of 0140978fcbaf05ef0a88c8cc9bcfc6da105d6933, it now installs and most of the functionality works in python3.3 and above.  I have not dealt with all of the `buffer` to `string` conversions yet, but when that's done it should finish up the remaining issues with connectivity via `xmlrpc` and `ssh`.  It's very raw at this point… tests pass, however.

**Comment by mmckerns** [comment]
This is a duplicate of #1.  Closing it.
